crates/bevy_reflect/src/tuple_struct.rs

```diff
@@ -398,6 +399,15 @@ impl Debug for DynamicTupleStruct {
     }
 }
 
+impl From<DynamicTuple> for DynamicTupleStruct {
+    fn from(value: DynamicTuple) -> Self {
+        Self {
+            represented_type: None,
+            fields: Box::new(value).drain(),
+        }
+    }
+}
+
```

Comment:
yet another place where we might be better off merging `Tuple` and `TupleStruct` :thinking:

Reply:
Personally, I still think we should keep the concepts separate, both for user clarity and the potential to allow their definitions to diverge as needed. If anything, we might try to reduce some code duplication by making better use of macro_rules.